docs/resources/server/enchantments.md

```diff
@@ -0,0 +1,319 @@
+Enchantments are special effects that can be applied to tools and other items. As of 1.21, enchantments are stored on items as [Data Components], are defined in JSON, and are comprised of Enchantment Effect Components.
+
+A new enchantment can be added by creating a JSON file in your namespace's `enchantment` datapack subfolder. For example, to create an enchantment called `examplemod:example_enchant`, one would create a file `data/examplemod/enchantment/example_enchantment.json`. 
+
+## Enchantment JSON Format
+- `description`: The [text component] that will be used as the in-game name of the enchantment.
+- `supported_items`: The items or item tag that this enchantment can be applied to.
+- `primary_items`: (Optional) The items or item tag that this enchantment appears for in the enchanting table. If left unspecified, this is the same as `supported_items`.
+- `exclusive_set`: One or a list of enchantments or enchantment tags that defines mutually exclusive (or 'incompatible') enchantments to this one.
+- `weight`: The likelihood that this enchantment will appear in the Enchanting Table from 1 to 1024.
+- `max_level`: The maximum level of this enchantment from 1 to 255.
+- `min_cost`: The minimum cost of this enchantment in levels. The actual cost will be randomized between this and `max_cost`.
+    - `base`: The base cost of the enchantment at level 1.
+    - `per_level_above_first`: The cost increase per enchantment level.
+- `max_cost`: The maximum cost of this enchantment.
+    - `base`: As above.
+    - `per_level_above_first`: As above.
+- `anvil_cost`: The cost this enchantment adds to repairing an item in the Anvil in levels. This cost is multiplied by enchantment level and halved if the enchant is on a book.
+- `slots`: (Optional) A list of slot groups this enchantment provides effects in. Can include any of: `any`, `hand`, `mainhand`, `offhand`, `armor`, `feet`, `legs`, `chest`, `head`, `body`
+- `effects`: The effects this enchantment provides as a map of Enchantment Effect Components.
+
+## Enchantment Effect Components
+Enchantment Effect Components are specially-registered [Data Components] that determine what an enchantment does. Enchantment Effect Component Types must be [registered] to `BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE`, but are otherwise identical to other Data Components. For example, one could register an Enchantment Effect Component that can store an `ExampleEffectData` object as follows:
+
+```java
+// In some registration class
+public static final DeferredRegister<DataComponentType<?>> ENCHANTMENT_COMPONENT_TYPES = DeferredRegister.create(BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE, "examplemod");
+
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<ExampleEffectData>> EXAMPLE =
+    ENCHANTMENT_COMPONENT_TYPES.register("example",
+        () -> DataComponentType.<ExampleEffectData>builder()
+            .persistent(ExampleEffectData.CODEC)
+            .build());
+```
```

Comment:
I've decided to split the built-in component types into their own page, following the example of the Recipes and Data Maps sections. I think this should help the main Enchantments page flow better, since it doesn't have to be padded out by the list of already existing components.